### PR TITLE
 Added length check in copy_within's fast path

### LINKED ISSUE #4146: Assertion 'ecma_get_value_type_field (value) == ECMA_TYPE_DIRECT in ecma_copy_value
###### JerryScript revision
086d1cc127f12e31ed315b41ca59ec0c3e88c7f4
(latest master - 2020.08.18.)

###### Build platform
Ubuntu 18.04.5 LTS (Linux 4.15.0-112-generic x86_64)

###### Build steps
tools/build.py --debug --logging=on --error-messages=on --line-info=on

###### Test case
found by test262-esnext built-ins/Array/prototype/copyWithin/coerced-values-start-change-target.js

little bit smaller test case (without test harness files)
```
function longDenseArray(){
        var a = [0];
        for(var i = 0; i < 1024; i++){
                a[i] = i;
        }
        return a;
}

function shorten(){
        currArray.length = 20;
        return 1;
}

var array = longDenseArray();
array.length = 20;
for(var i = 0; i < 19; i++){
        array[i+1000] = array[i+1];
}

var currArray = longDenseArray();
currArray.copyWithin(1000, {valueOf: shorten})
```

###### Output
```
ICE: Assertion 'ecma_get_value_type_field (value) == ECMA_TYPE_DIRECT || ecma_get_value_type_field (value) == ECMA_TYPE_DIRECT_STRING' failed at /home/oszi/jerryscript/jerry-core/ecma/base/ecma-helpers-value.c(ecma_copy_value):920.
Error: ERR_FAILED_INTERNAL_ASSERTION
```

###### Expected behavior
not assert

### PR DESCRIPTION
**Pull request by kisbg**
fixes #4146

JerryScript-DCO-1.0-Signed-off-by: bence gabor kis kisbg@inf.u-szeged.hu

### CODE REVIEW

File: .github/workflows/gh-actions.yml [hunk 1]
```diff
@@ -114,7 +114,7 @@ jobs:
     runs-on: ubuntu-18.04
     env:
       ASAN_OPTIONS: detect_stack_use_after_return=1:check_initialization_order=true:strict_init_order=true
-      SKIP_LIST: --skip-list=parser-oom.js,parser-oom2.js,stack-limit.js,regression-test-issue-2190.js,regression-test-issue-2258-2963.js,regression-test-issue-2448.js,regression-test-issue-2905.js,regression-test-issue-3785.js
+      SKIP_LIST: --skip-list=parser-oom.js,parser-oom2.js,stack-limit.js,regression-test-issue-2190.js,regression-test-issue-2258-2963.js,regression-test-issue-2448.js,regression-test-issue-2905.js,regression-test-issue-3785.js,regression-test-issue-4146.js
```
Comment (rerobika) [inline .github/workflows/gh-actions.yml:117]: This definitely should not be skiplisted.

File: tests/jerry/es.next/regression-test-issue-4146.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+function longDenseArray(){
+        var a = [0];
+        for(var i = 0; i < 1024; i++){
```
Comment (rerobika) [inline tests/jerry/es.next/regression-test-issue-4146.js:17]: reduce the 1024 to smaller value.

**Review by rerobika — CHANGES_REQUESTED** [review]

File: .github/workflows/gh-actions.yml [hunk 3]
```diff
@@ -135,4 +135,3 @@ jobs:
         run: sudo apt-get install gcc-multilib
       - name: UBSAN - jerry-tests
         run: $RUNNER --jerry-tests $SKIP_LIST $BUILD_OPTIONS -q
-
```
Comment (rerobika) [inline .github/workflows/gh-actions.yml:138]: Please revert this change.

File: tests/jerry/es.next/regression-test-issue-4146.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+var array = longDenseArray();
+array.length = 20;
+for (var i = 0; i < 19; i++){
+        array[i+1000] = array[i+1];
+}
+
+var currArray = longDenseArray();
+assert (currArray.copyWithin(200, {valueOf: shorten}).length == 20)
```
Comment (rerobika) [inline tests/jerry/es.next/regression-test-issue-4146.js:35]: Also please add an assert for the content of the array as well.

**Review by rerobika — CHANGES_REQUESTED** [review]

File: tests/jerry/es.next/regression-test-issue-4146.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+function shorten(){
+        currArray.length = 20;
+        return 1;
+}
+
+var currArray = longDenseArray();
+assert (currArray.copyWithin (200, {valueOf: shorten}).length == 20)
+var array = [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19];
+print (currArray.copyWithin(200, {valueOf: shorten}).equals (array))
```
Comment (rerobika) [inline tests/jerry/es.next/regression-test-issue-4146.js:49]: print -> assert

**Review by rerobika — APPROVED** [review]
LGTM, just please squash the commits.

**Review by galpeter — APPROVED** [review]
lgtm

**Pull request merged**